recipes/spark.rb

```diff
@@ -48,29 +48,10 @@
 end
 
 unless node['spark']['release']['install'].to_s == 'false'
-
-  # Spark binary compatibility matrix
-  case node['hadoop']['distribution']
-  when 'cdh'
-    spark_release = if node['hadoop']['distribution_version'].to_i == 4
-                      'cdh4'
-                    elsif node['hadoop']['distribution_version'].to_f == 5.0 || node['hadoop']['distribution_version'].to_f == 5.1
-                      'hadoop2.3'
-                    else
-                      'hadoop2.4'
-                    end
-  when 'hdp'
-    spark_release = if node['hadoop']['distribution_version'] == '2' || node['hadoop']['distribution_version'].to_f == 2.1
-                      'hadoop2.4'
-                    else
-                      ### TODO: HDP 2.0 is Hadoop 2.2... does 2.3 work?
-                      'hadoop2.3'
-                    end
-  end
+  spark_release = node['spark']['release']['hadoop_version']
 
   remote_file "#{node['spark']['release']['install_path']}/spark-#{node['spark']['release']['version']}-bin-#{spark_release}.tgz" do
     source "http://d3kbcqa49mib13.cloudfront.net/spark-#{node['spark']['release']['version']}-bin-#{spark_release}.tgz"
-    checksum node['spark']['release']['checksum']
```

Comment:
It's used by `remote_file` to ensure that we downloaded the correct file. The intention was for the user to modify this attribute at the same time as `node['spark']['release']['version']` or we would need to keep a compatibility matrix in the cookbook, which are a pain to maintain. I decided to go with this and risk a failed download, but they're coming from CloudFront, which has about as good reliability as you can get.